core/src/main/java/io/grpc/ServerInterceptors.java

```diff
@@ -80,6 +84,69 @@ public static ServerServiceDefinition intercept(ServerServiceDefinition serviceD
     return serviceDefBuilder.build();
   }
 
+  /**
+   * Create a new {@code ServerServiceDefinition} whose {@link MethodDescriptor} serializes to
+   * and from byte[] for all methods.  The {@code ServerCallHandler} created will automatically
+   * convert back to the original types for request and response before calling the existing
+   * {@code ServerCallHandler}.  Calling this method combined with the intercept methods will
+   * allow the developer to choose whether to intercept messages of byte[], or the modeled types
+   * of their application.
+   *
+   * @param serviceDef the service definition to convert messages to byte[]
+   * @return a wrapped version of {@code serviceDef} with the byte[] conversion applied.
+   */
+  public static ServerServiceDefinition useByteArrayMessages(
+      final ServerServiceDefinition serviceDef) {
+    final MethodDescriptor.Marshaller<byte[]> marshaller =
+        new MethodDescriptor.Marshaller<byte[]>() {
+      @Override
+      public InputStream stream(final byte[] value) {
+        return new ByteArrayInputStream(value);
+      }
+
+      @Override
+      public byte[] parse(final InputStream stream) {
+        try {
+          return ByteStreams.toByteArray(stream);
+        } catch (IOException e) {
+          throw new RuntimeException(e);
+        }
+      }
+    };
+
+    return useMarshalledMessages(serviceDef, marshaller);
+  }
+
+  /**
+   * Create a new {@code ServerServiceDefinition} whose {@link MethodDescriptor} serializes to
+   * and from T for all methods.  The {@code ServerCallHandler} created will automatically
+   * convert back to the original types for request and response before calling the existing
+   * {@code ServerCallHandler}.  Calling this method combined with the intercept methods will
+   * allow the developer to choose whether to intercept messages of T, or the modeled types
+   * of their application.  This can also be chained to allow for interceptors to handle messages
+   * as multiple different T types within the chain if the added cost of serialization is not
+   * a concern.
+   *
+   * @param serviceDef the service definition to convert messages to T
+   * @return a wrapped version of {@code serviceDef} with the T conversion applied.
+   */
+  public static <T> ServerServiceDefinition useMarshalledMessages(
```

Comment:
This looks fine, but let's annotation it with `@ExperimentalApi` since it's unclear how much use it will get.